stingray/tests/test_base.py

```diff
@@ -0,0 +1,216 @@
+
+        _check_equal(so, new_so)
+
+    @pytest.mark.skipif("not _HAS_PANDAS")
+    def test_pandas_roundtrip(self):
+        so = copy.deepcopy(self.sting_obj)
+        assert np.allclose(so.guefus, [4, 5, 2])
+        so.guefus = np.random.randint(0, 4, 3)
+        ts = so.to_pandas()
+        new_so = DummyStingrayObj.from_pandas(ts)
+
+        _check_equal(so, new_so)
+
+    def test_astropy_roundtrip_empty(self):
+        # Set an attribute to a DummyStingrayObj. It will *not* be saved
+        so = DummyStingrayObj([])
+        ts = so.to_astropy_table()
+        new_so = DummyStingrayObj.from_astropy_table(ts)
+        assert new_so.guefus is None or len(new_so.guefus) == 0
+
+    @pytest.mark.skipif("not _HAS_XARRAY")
+    def test_xarray_roundtrip_empty(self):
+        so = DummyStingrayObj([])
+        ts = so.to_xarray()
+        new_so = DummyStingrayObj.from_xarray(ts)
+
+        assert new_so.guefus is None or len(new_so.guefus) == 0
+
+    @pytest.mark.skipif("not _HAS_PANDAS")
+    def test_pandas_roundtrip_empty(self):
+        so = DummyStingrayObj([])
+        ts = so.to_pandas()
+        new_so = DummyStingrayObj.from_pandas(ts)
+
+        assert new_so.guefus is None or len(new_so.guefus) == 0
+
+    @pytest.mark.skipif("not _HAS_H5PY")
+    def test_hdf_roundtrip(self):
+        so = copy.deepcopy(self.sting_obj)
+        so.write("dummy.hdf5")
+        new_so = so.read("dummy.hdf5")
+        os.unlink("dummy.hdf5")
+
+        _check_equal(so, new_so)
+
+    @pytest.mark.skipif("not _HAS_H5PY")
+    def test_hdf_roundtrip_give_old__format(self):
+        so = copy.deepcopy(self.sting_obj)
+        so.guefus = np.random.randint(0, 4, 3)
+        with pytest.warns(DeprecationWarning):
+            so.write("dummy.hdf5", format_="hdf5")
+        with pytest.warns(DeprecationWarning):
+            new_so = DummyStingrayObj.read("dummy.hdf5", format_="hdf5")
+        os.unlink("dummy.hdf5")
+
+        _check_equal(so, new_so)
+
+    def test_file_roundtrip_fits(self):
+        so = copy.deepcopy(self.sting_obj)
+        so.guefus = np.random.randint(0, 4, 3)
+        so.panesapa = np.random.randint(5, 9, (6, 2))
+        so.write("dummy.fits")
+        new_so = DummyStingrayObj.read("dummy.fits")
+        os.unlink("dummy.fits")
+        # panesapa is invalid for FITS header and got lost
+        assert not hasattr(new_so, "panesapa")
+        new_so.panesapa = so.panesapa
+        _check_equal(so, new_so)
+
+    @pytest.mark.parametrize("fmt", ["pickle", "ascii", "ascii.ecsv"])
+    def test_file_roundtrip(self, fmt):
+        so = copy.deepcopy(self.sting_obj)
+        so.guefus = np.random.randint(0, 4, 3)
+        so.panesapa = np.random.randint(5, 9, (6, 2))
+        so.write(f"dummy.{fmt}", fmt=fmt)
+        new_so = DummyStingrayObj.read(f"dummy.{fmt}", fmt=fmt)
+        os.unlink(f"dummy.{fmt}")
+
+        _check_equal(so, new_so)
+
+
+class TestStingrayTimeseries:
+    @classmethod
```

Comment:
Should there be tests in this test class for some of the other functionality in the time series? E.g. the time shifting and so on that you moved there?